pkg/comp-functions/functions/vshnkeycloak/deploy.go

```diff
@@ -216,13 +230,115 @@ func newValues(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VS
 		return values, fmt.Errorf("cannot fetch nodeSelector from the composition config: %w", err)
 	}
 
+	extraEnvMap := []map[string]any{
+		{
+			"name":  "KEYCLOAK_ADMIN",
+			"value": "admin",
+		},
+		{
+			"name": "KEYCLOAK_ADMIN_PASSWORD",
+			"valueFrom": map[string]any{
+				"secretKeyRef": map[string]any{
+					"name": adminSecret,
+					"key":  adminPWSecretField,
+				},
+			},
+		},
+		{
+			"name":  "KC_DB_URL_PROPERTIES",
+			"value": "?sslmode=verify-full&sslrootcert=/certs/ca.crt",
+		},
+		{
+			"name": "JAVA_OPTS_APPEND",
+			"value": `-Djava.awt.headless=true
+-Djgroups.dns.query={{ include "keycloak.fullname" . }}-headless`,
+		},
+	}
+
+	extraEnv, err := toYAML(extraEnvMap)
+	if err != nil {
+		return nil, err
+	}
+
+	extraVolumesMap := []map[string]any{
+		{
+			"name":     "custom-providers",
+			"emptyDir": nil,
+		},
+		{
+			"name":     "custom-setup",
+			"emptyDir": nil,
+		},
+		{
+			"name": "keycloak-dist",
+		},
+		{
+			"name": "postgresql-certs",
+			"secret": map[string]any{
+				"secretName":  pgSecretName,
+				"defaultMode": 420,
+			},
+		},
+	}
+
+	extraVolumes, err := toYAML(extraVolumesMap)
+	if err != nil {
+		return nil, err
+	}
+
+	extraVolumeMountsMap := []map[string]any{
+		{
+			"name":      "custom-providers",
+			"mountPath": "/opt/keycloak/providers",
+		},
+		{
+			"name":      "custom-setup",
+			"mountPath": "/opt/keycloak/setup",
+		},
+		{
+			"name":      "postgresql-certs",
+			"mountPath": "/certs/",
+		},
+	}
+
+	extraVolumeMounts, err := toYAML(extraVolumeMountsMap)
+	if err != nil {
+		return nil, err
+	}
+
 	values = map[string]any{
-		"replicaCount": "1",
-		"production":   false, // TODO: we can enable production mode with proper TLS only
-		"auth": map[string]any{
-			"adminUser":         "admin",
-			"existingSecret":    adminSecret,
-			"passwordSecretKey": adminPWSecretField,
+		"extraEnv":          extraEnv,
+		"extraVolumes":      extraVolumes,
+		"extraVolumeMounts": extraVolumeMounts,
+
+		"command": []string{
+			"/opt/keycloak/bin/kc-with-setup.sh",
+			"--verbose",
+			"start",
+			"--http-enabled=true",
+			"--http-port=8080",
+			"--hostname-strict=false",
+			"--hostname-strict-https=false",
+			"--spi-events-listener-jboss-logging-success-level=info",
+			"--spi-events-listener-jboss-logging-error-level=warn",
+		},
+		"image": map[string]any{
+			"repository": "docker-registry.inventage.com:10121/keycloak-competence-center/keycloak-managed",
```

Comment:
Would it make sense, to make this configurable and not hardcoded?

Reply:
Did Inventage say anything about changing the repository?

We also have hardcoded charts/images for other services though.

Reply:
No they didn't. I just thought it might make it easier to change it if we have to. But fine by me to leave it as is.